R/eval_forecasts.R

```diff
@@ -140,13 +158,23 @@
 
 eval_forecasts <- function(data,
                            by = c("model"),
+                           summarise_by = NULL,
                            summarised = TRUE,
+                           quantiles = c(),
+                           sd = FALSE,
                            pit_arguments = list(plot = FALSE),
                            interval_score_arguments = list()) {
 
+
+  # preparations ---------------------------------------------------------------
   data.table::setDT(data)
 
-  ## check if predictions are integer, continuous, etc.
+  if (is.null(summarise_by)) {
```

Comment:
Suggest if can moving to the missing set up rather than a null check (missing(summarise_by)) as it seems a bit more elegant

Reply:
will do, thanks :)

Reply:
just thought about it again. Can you explain what the rationale behind the missing set up is and why it is more elegant? My thought was that setting it to NULL would be less confusing to users as they don't believe they necessarily have to set the argument to something

Reply:
because then it needs no default so the code is cleaner

Reply:
So this is with missing: https://github.com/epiforecasts/EpiNow2/blob/master/R/estimate_infections.R

and this with NULL: https://github.com/epiforecasts/EpiNow/blob/master/R/estimate_R0.R

Reply:
but it would litterally just be this since I need `summarise_by` later on? or am I missing something? 
![image](https://user-images.githubusercontent.com/37978797/88398681-020f5780-cdc6-11ea-8a63-7c6554cc912a.png)

Reply:
yes and changing the default of summarise_by in the function def

Reply:
you wouldn't need any of these empty defaults you have used: https://github.com/epiforecasts/scoringutils/blob/8d27cf08fcf6101ef381603707ab200022a2dbef/R/eval_forecasts.R#L160

Reply:
ah can I just write summarise_by = by in the function def? in that case I wouldn't even need the missing(summarise_by) at all...

Reply:
interesting - I think I can just write that in the function def and it works. the reason to have summarise_by is that sometimes you want to summarise over different categories then you want to score over. So you e.g. have one score for each region and then summarise over these regions --> this way you get quantiles that you can use for plotting.